arbnode/node.go

```diff
@@ -794,21 +797,53 @@ func createNodeImpl(
 	if config.Wasm.RootPath != "" {
 		nitroMachineConfig.RootPath = config.Wasm.RootPath
 	} else {
-		execfile, err := os.Executable()
-		if err != nil {
-			panic(err)
+		_, thisFile, _, ok := runtime.Caller(0) // <project>/arbnode/node.go
+		if !ok {
+			panic("failed to find root path")
 		}
-		targetDir := filepath.Dir(filepath.Dir(execfile))
-		nitroMachineConfig.RootPath = filepath.Join(targetDir, "machines")
+		projectDir := filepath.Dir(filepath.Dir(thisFile))
+		nitroMachineConfig.RootPath = filepath.Join(filepath.Join(projectDir, "target"), "machines")
```

Comment:
Looks like this won't work in the docker image, directory is `/home/user/target/machines` and the nitro executable is at `/usr/local/bin/nitro`.

How about storing in `~/.arbitrum/<chain>/machines/`, can be copied into docker image, and download automatically if it doesn't exist.

Reply:
As a stop-gap, I've added a `FindMachineDir()` that looks for the machines in those and other places.

I like that idea though, let's talk about it during standup :)

Reply:
If we make a standard place, I think we should just do `~/.arbitrum/machines` probably since they're hash addressed so they don't need to be chain specific